modules/openapi-generator/src/main/resources/ruby-client/faraday_api_client.mustache

```diff
@@ -0,0 +1,392 @@
+=begin
+{{> api_info}}
+=end
+
+require 'date'
+require 'faraday'
+require 'json'
+require 'logger'
+require 'tempfile'
+
+module {{moduleName}}
+  class ApiClient
+    # The Configuration object holding settings to be used in the API client.
+    attr_accessor :config
+
+    # Defines the headers to be used in HTTP requests of all API calls by default.
+    #
+    # @return [Hash]
+    attr_accessor :default_headers
+
+    # Initializes the ApiClient
+    # @option config [Configuration] Configuration for initializing the object, default to Configuration.default
+    def initialize(config = Configuration.default)
+      @config = config
+      @user_agent = "{{#httpUserAgent}}{{{.}}}{{/httpUserAgent}}{{^httpUserAgent}}OpenAPI-Generator/#{VERSION}/ruby{{/httpUserAgent}}"
+      @default_headers = {
+        'Content-Type' => 'application/json',
+        'User-Agent' => @user_agent
+      }
+    end
+
+    def self.default
+      @@default ||= ApiClient.new
+    end
+
+    # Call an API with given options.
+    #
+    # @return [Array<(Object, Integer, Hash)>] an array of 3 elements:
+    #   the data deserialized from response body (could be nil), response status code and response headers.
+    def call_api(http_method, path, opts = {})
+      connection = Faraday.new(:url => config.base_url) do |conn|
```

Comment:
@dkliban  Thanks so much for this PR!
I wonder if you have considered how to add middlewares here to fully use the power of Faraday. For instance one could add caching or tracing or whatever in here. I think that's one of the best things about moving to Faraday